### ISSUE TITLE
prefer-optional-chain false positive

### ISSUE THREAD

**Issue by camc314** [oxc-project/tsgolint#585]
```ts
declare const options: {watch?: string | boolean | (string | boolean)[] | undefined};

export const w =  !(options.watch === true || options.watch === "true")
```

1. Same property compared to multiple literals

```typescript
type NotificationType = "banner" | "psa" | "error" | "message";

const type: NotificationType = "banner";

// Flagged incorrectly
if (type === "banner" || type === "psa") {
  console.log("is banner or psa");
}
```

2. `instanceof` checks

```typescript
class Foo {}
class Bar {}

const value: Foo | Bar | null = new Foo();

// Flagged incorrectly
if (value instanceof Foo || value instanceof Bar) {
  console.log("is Foo or Bar");
}
```

3. Explicit nullish equality checks

```typescript
declare const language: string | undefined | null;

// Flagged incorrectly
if (language === undefined || language === null) {
  return "default";
}
```

4. Boolean property equality checks

```typescript
interface User {
  verified?: boolean;
}

const user: User = {};

// Flagged incorrectly
if (user.verified === false || user.verified === undefined) {
  console.log("not verified");
}
```

5. Type narrowing + property access

```typescript
interface Event {
  keySpacing: "toolong" | number[];
}

const event: Event = { keySpacing: [1, 2, 3] };

// Flagged incorrectly
if (event.keySpacing !== "toolong" && event.keySpacing.length > 0) {
  console.log("has key spacing data");
}
```

6. Nullish guard + length comparison

```typescript
const items: string[] | undefined = [];

// Flagged incorrectly
if (items !== undefined && items.length > 0) {
  console.log("has items");
}
```

7. Mixed conditions with different properties

```typescript
interface KeyboardEvent {
  code: string;
  key: string;
}

const event: KeyboardEvent = { code: "", key: "a" };

// Flagged incorrectly
if (event.code === "" || event.code === undefined || event.key === "Unidentified") {
  console.log("invalid event");
}
```

8. Already using optional chaining

```typescript
const element = document.activeElement;

// Flagged incorrectly
const isInteractive =
  document.activeElement?.tagName === "INPUT" ||
  document.activeElement?.tagName === "TEXTAREA" ||
  document.activeElement?.tagName === "SELECT";
```

The above cases report errors, but should not

**Comment by camchenry** [comment]
Added passing test cases for all of these in https://github.com/oxc-project/tsgolint/pull/596.
